src/controllers/otpauth.controllers.ts

```diff
@@ -0,0 +1,82 @@
+import { Request, Response } from "express";
+import randomstring from "randomstring";
+import Jwt from "jsonwebtoken";
+import { config } from "dotenv";
+import { sendOTPCode } from "../utils/mailer";
+
+config();
+
+export interface dataInfo {
+  body: {
+    email: string;
+    otp: string;
+  };
+}
+interface userData {
+  dataValues: {
+    firstName: string;
+    lastName: string;
+    email: string;
+  };
+}
+// Controller to send the OTP to the server
+const sendOtp = (req: Request, res: Response) => {
+  const user = req.user as userData;
+
+  const { email } = user.dataValues;
+
+  // Generate a unique 6 number digit
+  const code = randomstring.generate({
+    length: 6,
+    charset: "numeric"
+  });
+
+  // The body for JWT token data
+  const body = {
+    email,
+    otp: code
+  };
+
+  // create a new JWT token
+  const token = Jwt.sign({ body }, process.env.JWT_SECRET as string, {
+    expiresIn: "60s"
+  });
+
+  sendOTPCode(email, code);
+
+  res.status(201).json({
+    message: "Verification code sent successfully",
+    otpToken: `${token}`
+  });
+};
+
+// Controller to verify the token
+const verifyOtp = (req: Request, res: Response) => {
+  const { token } = req.params;
+  const { otp } = req.body;
+  if (!otp) {
+    return res.status(404).json({ message: "Invalid OTP" });
+  }
+  // verify the token from the params
+  Jwt.verify(token, process.env.JWT_SECRET as string, (err, decoded) => {
+    if (err) {
```

Comment:
This code is duplicated across your codebase. You should have a functions or middleware that encodes, decodes and verifies the token respectively.